### PR TITLE
plocate: init at 1.1.7

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

#124081

Been using a slighly earlier version as a daily driver for a few months.

Cc: @KenMacD

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -25356,6 +25356,8 @@ in
 
   mlocate = callPackage ../tools/misc/mlocate { };
 
+  plocate = callPackage ../tools/misc/plocate { };
```
Comment (06kellyjac) [inline pkgs/top-level/all-packages.nix:25359]: Personally I'd prefer this either go somewhere more alphabetically appropriate or remove the newlines between `mlocate` and `plocate` like so:

```suggestion
  mlocate = callPackage ../tools/misc/mlocate { };
  plocate = callPackage ../tools/misc/plocate { };
```

Comment (06kellyjac) [reply pkgs/top-level/all-packages.nix:25359]: I'm aware which is why I gave the two suggestions, one not involving moving it alphabetically.
Or you can leave it as is :shrug:

File: pkgs/tools/misc/plocate/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+, liburing
+, zstd
+}:
+
+stdenv.mkDerivation rec {
+  pname = "plocate";
+  version = "1.1.7";
+
+  src = fetchgit {
+    url = "http://git.sesse.net/plocate";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/plocate/default.nix:17]:
```suggestion
    url = "https://git.sesse.net/plocate";
```

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/top-level/all-packages.nix:25359]: `plocate` is a different software; there is no reason to put them without whitespace.

Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:25359]: all-packages is not sorted at all. In the grandscheme of things this is irrelevant.

**Comment by mkgvt** [comment]
@peterhoeg I built plocate from the default.nix and it couldn't find liburing.so.1:
```
$ curl -LO https://raw.githubusercontent.com/NixOS/nixpkgs/master/pkgs/tools/misc/plocate/default.ni
$ nix-build -E 'with import <nixpkgs> {}; callPackage ./default.nix
$ result/bin/plocate
plocate: error while loading shared libraries: liburing.so.1: cannot open shared object file: No such file or directory
```
Indeed, there doesn't appear to be an entry in the linker table. 
```
$ ldd result/bin/plocate
	linux-vdso.so.1 (0x00007ffe3dead000)
	liburing.so.1 => not found
	libzstd.so.1 => /nix/store/30gam09v1xiz8lfghwry1cm61834lsl5-zstd-1.4.5/lib/libzstd.so.1 (0x00007fd700f21000)
	libstdc++.so.6 => /nix/store/c10296m7xgm3ksibcklb2xf48jr635x3-gcc-9.3.0-lib/lib/libstdc++.so.6 (0x00007fd700d40000)
	libgcc_s.so.1 => /nix/store/0c7c96gikmzv87i7lv3vq5s1cmfjd6zf-glibc-2.31-74/lib/libgcc_s.so.1 (0x00007fd700d26000)
	libpthread.so.0 => /nix/store/0c7c96gikmzv87i7lv3vq5s1cmfjd6zf-glibc-2.31-74/lib/libpthread.so.0 (0x00007fd700d05000)
	libc.so.6 => /nix/store/0c7c96gikmzv87i7lv3vq5s1cmfjd6zf-glibc-2.31-74/lib/libc.so.6 (0x00007fd700b46000)
	/nix/store/0c7c96gikmzv87i7lv3vq5s1cmfjd6zf-glibc-2.31-74/lib/ld-linux-x86-64.so.2 => /nix/store/0c7c96gikmzv87i7lv3vq5s1cmfjd6zf-glibc-2.31-74/lib64/ld-linux-x86-64.so.2 (0x00007fd700fda000)
	libm.so.6 => /nix/store/0c7c96gikmzv87i7lv3vq5s1cmfjd6zf-glibc-2.31-74/lib/libm.so.6 (0x00007fd700a03000)
```
I don't see anything obviously in default.nix and liburing.so.1 is in /nix/store:
```
$ \ls /nix/store/ | grep plocate
24wairnpq6wshbdnjb0vv5hrjrrkjgr6-plocate-1.1.7/
i58lf7jq1ll62lhvdwzvchpr32fh7vzl-plocate-1.1.7.drv
x8i4pld8xrihszp07r4ii7q0q1899d8a-plocate/
ypfcd3g740j8w59rnlyn5vvk0yqr6vah-plocate.drv
$ less /nix/store/i58lf7jq1ll62lhvdwzvchpr32fh7vzl-plocate-1.1.7.drv  # /nix/store/2hml962j3mzyqchn2kw0gzm552dpry5f-liburing-0.7.drv
$ less /nix/store/2hml962j3mzyqchn2kw0gzm552dpry5f-liburing-0.7.drv  # /nix/store/jr91xav9iy628hx9f0znw7l0z6yj7j6v-liburing-0.7-lib
$ \ls /nix/store/jr91xav9iy628hx9f0znw7l0z6yj7j6v-liburing-0.7-lib/lib/liburing.so.1*
/nix/store/jr91xav9iy628hx9f0znw7l0z6yj7j6v-liburing-0.7-lib/lib/liburing.so.1
/nix/store/jr91xav9iy628hx9f0znw7l0z6yj7j6v-liburing-0.7-lib/lib/liburing.so.1.0.7
```
I wonder why it wasn't entered into the linker table?

**Comment by peterhoeg** [comment]
This is what I get on a checkout of master (same result on unstable):

$ nix-build -A plocate . && ldd ./result/bin/plocate
/nix/store/fxc0fm4k1mxww26qhhb2yqz16scd8ki3-plocate-1.1.7
        linux-vdso.so.1 (0x00007ffda2b9c000)
        liburing.so.2 => /nix/store/2zspd3vxdy5zbnnwx6j5r93m8nkn3dyh-liburing-2.0/lib/liburing.so.2 (0x00007fd9c6cf4000)
        libzstd.so.1 => /nix/store/jnwndjq8n8kmvrcad5ar7mhnn36y0fi1-zstd-1.5.0/lib/libzstd.so.1 (0x00007fd9c6c0b000)
        libstdc++.so.6 => /nix/store/d9mfvhvksvsx0ar8pmxwhgwd9my045rp-gcc-10.3.0-lib/lib/libstdc++.so.6 (0x00007fd9c6a36000)
        libgcc_s.so.1 => /nix/store/adxc893j47gxx3xjw403zdf0liiddvw2-glibc-2.32-48/lib/libgcc_s.so.1 (0x00007fd9c6a1c000)
        libpthread.so.0 => /nix/store/adxc893j47gxx3xjw403zdf0liiddvw2-glibc-2.32-48/lib/libpthread.so.0 (0x00007fd9c69fb000)
        libc.so.6 => /nix/store/adxc893j47gxx3xjw403zdf0liiddvw2-glibc-2.32-48/lib/libc.so.6 (0x00007fd9c6838000)
        /nix/store/adxc893j47gxx3xjw403zdf0liiddvw2-glibc-2.32-48/lib/ld-linux-x86-64.so.2 => /nix/store/adxc893j47gxx3xjw403zdf0liiddvw2-glibc-2.32-48/lib64/ld-linux-x86-64.so.2 (0x00007fd9c6cfd000)
        libm.so.6 => /nix/store/adxc893j47gxx3xjw403zdf0liiddvw2-glibc-2.32-48/lib/libm.so.6 (0x00007fd9c66f5000)

Can you try that?

**Comment by mkgvt** [comment]
It builds and contains a linker entry for liburing: 
```
$ nix-build https://github.com/NixOS/nixpkgs/archive/master.tar.gz -A plocate
...
$ ldd ./result/bin/plocate
	linux-vdso.so.1 (0x00007ffcea561000)
	liburing.so.2 => /nix/store/2zspd3vxdy5zbnnwx6j5r93m8nkn3dyh-liburing-2.0/lib/liburing.so.2 (0x00007f501e31b000)
	libzstd.so.1 => /nix/store/jnwndjq8n8kmvrcad5ar7mhnn36y0fi1-zstd-1.5.0/lib/libzstd.so.1 (0x00007f501e232000)
	libstdc++.so.6 => /nix/store/d9mfvhvksvsx0ar8pmxwhgwd9my045rp-gcc-10.3.0-lib/lib/libstdc++.so.6 (0x00007f501e05d000)
	libgcc_s.so.1 => /nix/store/adxc893j47gxx3xjw403zdf0liiddvw2-glibc-2.32-48/lib/libgcc_s.so.1 (0x00007f501e043000)
	libpthread.so.0 => /nix/store/adxc893j47gxx3xjw403zdf0liiddvw2-glibc-2.32-48/lib/libpthread.so.0 (0x00007f501e022000)
	libc.so.6 => /nix/store/adxc893j47gxx3xjw403zdf0liiddvw2-glibc-2.32-48/lib/libc.so.6 (0x00007f501de5f000)
	/nix/store/adxc893j47gxx3xjw403zdf0liiddvw2-glibc-2.32-48/lib/ld-linux-x86-64.so.2 => /nix/store/0c7c96gikmzv87i7lv3vq5s1cmfjd6zf-glibc-2.31-74/lib64/ld-linux-x86-64.so.2 (0x00007f501e324000)
	libm.so.6 => /nix/store/adxc893j47gxx3xjw403zdf0liiddvw2-glibc-2.32-48/lib/libm.so.6 (0x00007f501dd1c000)
```
Looks like I shouldn't have tried to cherry pick plocate into 20.09 as it doesn't work.

**Pull request merged**